src/Sulu/Bundle/AdminBundle/Metadata/FormMetadata/Validation/BlockFieldMetadataValidator.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Metadata\FormMetadata\Validation;
+
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\FieldMetadata;
+
+class BlockFieldMetadataValidator implements FieldMetadataValidatorInterface
+{
+    const RESERVED_PROPERTY_NAMES = ['type', 'settings'];
+
+    public function validate(FieldMetadata $fieldMetadata): void
+    {
+        if ('block' !== $fieldMetadata->getType()) {
+            return;
+        }
+
+        foreach ($fieldMetadata->getTypes() as $type) {
+            foreach ($type->getItems() as $item) {
+                if (!$item instanceof FieldMetadata) {
+                    continue;
+                }
+
+                if (\in_array($item->getName(), static::RESERVED_PROPERTY_NAMES, true)) {
+                    throw new InvalidFieldMetadataException(
```

Comment:
I'd prefer to pass the information directly to the constructor, instead of generating the message everywhere the exception is created. Less code duplication and it would allow to retrieve the information from the exception in a `catch` block without parsing the exception message. So IMO something like this should be possible:

```php
try {
    throw new InvalidFieldMetadataException($fieldMetadata->getName(), $item->getName());
} catch (InvalidFieldMetadataException $e) {
    $e->getFieldName();
    $e->getPropertyName();
}
```

TBH I am also wondering why you have introduced another `Exception` instead of reusing the `ReservedPropertyNameException`. That one would also explain better what that exception is about.

Reply:
One reason why we decided to introduce this kind of validation is to be able to validate more than just the property name, right? So I guess there should be a generic exception for all validation errors. But I would be okay with the `ReservedPropertyNameException` if it would extend from the `InvalidFieldMetadataException`. But currently we don't get the form key in the validator and the `ReservedPropertyNameException` needs that one too.

Reply:
Yeah, but we could also introduce other more specific exceptions in the future. For me it would also be ok for the `ReservedPropertyNameException` to inherit from the `InvalidFieldMetadataException`.

But passing the form key is essential to this exception, because otherwise it is very hard for the developer to figure out which form is affected. If we introduce a new inheritance hierarchy I would even add that key to the `InvalidFieldMetadataException`.